hyper/http20/connection.py

```diff
@@ -272,13 +275,23 @@ def request(self, method, url, body=None, headers=None):
         # being sent in the wrong order, which can lead to the out-of-order
         # messages with lower stream IDs being closed prematurely.
         with self._write_lock:
+            # Unlike HTTP/1.1, HTTP/2 (according to RFC 7540) doesn't require
+            # to use absolute URI when proxying.
+
             stream_id = self.putrequest(method, url)
 
             default_headers = (':method', ':scheme', ':authority', ':path')
             for name, value in headers.items():
                 is_default = to_native_string(name) in default_headers
                 self.putheader(name, value, stream_id, replace=is_default)
 
+            # Append proxy headers.
+            if self.proxy_host and not self.secure:
+                proxy_headers = self.proxy_headers or {}
+                for name, value in proxy_headers.items():
+                    is_default = to_native_string(name) in default_headers
+                    self.putheader(name, value, stream_id, replace=is_default)
```

Comment:
It would be better to try to be a bit clever here. We can avoid duplicating the loop above by doing this:

```python
all_headers = headers.items()
if self.proxy_host and not self.secure:
    proxy_headers = self.proxy_headers or {}
    all_headers = itertools.chain(all_headers, proxy_headers.items())

for name, value in all_headers:
```

That'll keep a single loop, and generally seem a lot cleaner.